### DISCUSSION TITLE
LabeledSelect not importing CSS with CRA

### DISCUSSION THREAD

**Discussion by wsm95** [iTwin/iTwinUI#1194] [Q&A] [answered]
I wasn't quite sure if this was an issue with iTwinUI or something on my end with how CRA is configured, so I decided to make a discussion post rather a full blown issue.

I have been developing a new web app using CRA and noticed when I tried to use the `LabeledSelect` component it was not styled correctly.  I went reproduce this issue using a Code Sandbox but could not reproduce it:

https://codesandbox.io/s/jovial-shadow-mxhoed?file=/src/App.jsx

I could get it to reproduce locally after create a very barebones CRA example:

![image](https://user-images.githubusercontent.com/11621478/230668853-58e4552e-8118-478b-9244-1a071d3157bd.png)
![image](https://user-images.githubusercontent.com/11621478/230668891-043ba19e-c439-4c90-9cc0-e11fda713ef1.png)

I think the issue comes from how the CSS is imported inside the `LabeledSelect` component, and the difference with how Code Sandbox and CRA import CSS. Code Sandbox looks to be importing all of the CSS from iTwinUI, while CRA looks to be only packaging the CSS that is directly used:

![image](https://user-images.githubusercontent.com/11621478/230669548-802a8ed0-65f9-48f5-a468-3b331eb0b79d.png)
![image](https://user-images.githubusercontent.com/11621478/230670135-1dc52a15-2b8d-4c1a-a524-3af1c91e685f.png)

I have found a working around by directly adding a reference directly to CSS file:

![image](https://user-images.githubusercontent.com/11621478/230671452-46e43a83-0d89-4d00-970c-3273085ff5bd.png)

I am not sure if this is the intended behavior of iTwinUI (importing css files directly) and I am wondering if there are any CRA configurations I should be aware of to solve this issue.

**Comment by mayank99** [comment, marked as answer]
~~It looks like some styles are still applying, so we might be missing an import somewhere inside `LabeledSelect`.~~ edit: whoops, I see you already mentioned you worked around it with a manual import.

Yeah, we are missing an import for `input.css`. I've created a PR: https://github.com/iTwin/iTwinUI/pull/1195

**Reply by wsm95** [reply]
Thank you for the quick resolution and PR! Once that gets completed and released, I will test and make sure that officially resolves the issue.

**Reply by mayank99** [reply]
Now released in `@itwin/itwinui-react@2.8.1` 🚀

**Reply by mayank99** [reply]
@wsm95 Can you please let us know if the fix worked?

A couple more interesting notes I should probably have mentioned in my initial comment:
- When testing, make sure to also run the prod build. Often times, the dev server loads up every possible CSS file without treeshaking concerns.
- If you are using any other dependencies that make use of iTwinUI too, then you could end up with unexpected issues. For guidance on how to handle this, see this wiki page: [Version conflicts](https://github.com/iTwin/iTwinUI/wiki/Version-conflicts)
- In the next major version, we will likely remove all CSS imports and provide a single bundled CSS file that you can import manually. This will give you full control over the CSS, and you can make use of things like [cascade layers](https://css-tricks.com/css-cascade-layers/) to battle any specificity issues.
- Suggestion: If you're building a new app, please don't use CRA as it is no longer maintained. The best option for client-heavy apps today is to use Next.js or start with a Vite template if you want something more barebones. We have [examples](https://github.com/iTwin/iTwinUI/tree/main/playgrounds) of both (openable in stackblitz). For more, see the [new react docs](https://react.dev/learn/start-a-new-react-project).
